nlp/2021_06_29_long_range_transformers/LongRangeTransformers.ipynb

```diff
@@ -0,0 +1,1940 @@
+    "    train_ds = train_ds.map(\n",
+    "        tokenization, batched=True, batch_size=len(train_ds), remove_columns=['text'])\n",
+    "    test_ds = test_ds.map(tokenization, batched=True,\n",
+    "                          batch_size=len(test_ds),  remove_columns=['text'])\n",
+    "    train_ds.set_format(\n",
+    "        'torch', columns=['input_ids', 'attention_mask', 'label'])\n",
+    "    test_ds.set_format('torch', columns=[\n",
+    "                       'input_ids', 'attention_mask', 'label'])\n",
+    "\n",
+    "\n",
+    "    parameters = model.num_parameters()\n",
+    "    training_args = TrainingArguments(\n",
+    "        num_train_epochs=5,\n",
+    "        per_device_train_batch_size=4,\n",
+    "        gradient_accumulation_steps=2,\n",
+    "        per_device_eval_batch_size=4,\n",
+    "        evaluation_strategy=\"steps\",\n",
+    "        eval_steps=100, \n",
+    "        disable_tqdm=False,\n",
+    "        load_best_model_at_end=True,\n",
+    "        metric_for_best_model='accuracy', # default value is validation loss, we want the model with the highest accuracy \n",
+    "        warmup_steps=200,\n",
+    "        logging_steps=100,\n",
+    "        learning_rate=5e-6,\n",
+    "        fp16=True,\n",
+    "        dataloader_num_workers=0,\n",
+    "        output_dir='./{}-classification-consumer-complaints-results'.format(\n",
+    "            identifier),\n",
+    "        run_name='{}-classification-consumer-complaints'.format(identifier),\n",
+    "        logging_dir='./{}-classification-consumer-complaints-logging'.format(\n",
+    "            identifier),\n",
+    "        save_strategy=\"steps\"\n",
+    "    )\n",
+    "\n",
+    "    trainer = Trainer(\n",
+    "        model=model,\n",
+    "        args=training_args,\n",
+    "        compute_metrics=compute_metrics,\n",
+    "        train_dataset=train_ds,\n",
+    "        eval_dataset=test_ds,\n",
+    "    )\n",
+    "    start_time = time.time()\n",
+    "    trainer.train()\n",
+    "    duration = time.time() - start_time\n",
+    "\n",
+    "    metrics = trainer.evaluate()\n",
+    "\n",
+    "    trainer.save_model(f\"models/{identifier}\")\n",
+    "    tokenizer.save_pretrained(f\"models/{identifier}\")\n",
+    "\n",
+    "    return trainer, parameters, metrics, duration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {
+    "id": "_VomBSwDejiy"
+   },
+   "outputs": [],
+   "source": [
+    "token_count_cutoff = IMDBDatasetTokenCountType.LONG # Run for both SHORT and LONG scenario"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "1JYNVDC8BSZW",
+    "outputId": "0ad30f96-d5b9-4ec9-d3a9-8aa3ee3683b9"
+   },
+   "outputs": [],
+   "source": [
+    "train_ds, test_ds = filter_train_test_dataset_with_tokencount(ds['train'], ds['test'], token_count_cutoff)\n",
```

Comment:
Don't really quite follow what we want to achieve here with the current description...
Why do we want equal n° of training samples?
Also: where do the values of 1270 and 1186 come from?

Reply:
To have a fair comparison when comparing the impact of the token length
So that the model trained on short token length isn't performing better because there are more data samples to train on, do you think this is a valid approach? Maybe I could restructure it better to clarify the incentive?